content/millennium/r4/request-and-response/communication.md

```diff
@@ -0,0 +1,245 @@
+* [Payload](https://hl7.org/FHIR/communication-definitions.html#Communication.payload)
+
+## Terminology Bindings
+
+<%= terminology_table(:communication, :r4) %>
+
+## Extensions
+
+### Custom Extensions
+
+All URLs for custom extensions are defined as `https://fhir-ehr.cerner.com/r4/StructureDefinition/{id}`
+
+ID             | Value\[x] Type      | Description
+---------------|---------------------|----------------------------------------------------------------------------------
+`reply-to`     | [`Reference`]       | Used to direct where replies to the communication should be sent. Must be a Group.
+`email-status` | [`CodeableConcept`] | Status of the electronic communication (in-progress, completed, received).
+
+## Search
+
+Search for communications that meet supplied query parameters:
+
+    GET /Communication?:parameters
+
+_Implementation Notes_
+
+* The Content of a communication will be returned through the binary resource. A reference to the payload will be provided in the Payload section.
+* Only 1000 elements max will be returned based on the date range.
+
+### Authorization Types
+
+<%= authorization_types(provider: true, patient: false, system: true) %>
+
+### Parameters
+
+ Name              | Required?                                                          | Type          | Description
+-------------------|--------------------------------------------------------------------|---------------|-----------------------------------------------------------------------------------------------------------------------------------------------
+ `_id`             | This or (`category`, `-email-status`, `recipient` and `received`) | [`token`]     | The logical resource id associated with the resource. Example: `489580643.0.-4.prsnl`
+ `category`        | (This, `recipient`, `received` and `-email-status`) or `_id`      | [`token`]     | A token for a [`CodeableConcept`] that points to the CodeableConcept used for Communication.category. Example: `http://terminology.hl7.org/CodeSystem/communication-category|notification`
+ `recipient`       | (This, `category`, `received` and `-email-status`) or `_id`       | [`reference`] | The recipient of the communication. Example: `Practitioner/3456783`
+ `received`        | (This, `category`, `recipient` and `-email-status`) or `_id`      | [`date`]      | Date range into which the communication falls. Example: `le2017-02-01T10:30:00Z`
+ `-email-status`   | (This, `category`, `recipient` and `received`) or `_id`           | [`token`]     | The status of the email. Example: `http://hl7.org/fhir/task-status|in-progress`
+
+
+ Notes:
+
+ * The `recipient` parameter must represent a Practitioner or a Group. Use the `:Group` modifier to search by Group.
+ * The `received` parameter may be provided up to two times, and must use the `ge` or `le` prefixes. When provided twice, the lower value must have a `ge` prefix and the higher value must have a `le` prefix. The `ge` prefix is only supported on one.
```

Comment:
The 'ge' prefix is only supported when provided once??

Reply:
ge prefix is only supported on one of the received parameters. We could probably just say it's required

Reply:
Can the received simply be a date?  If so, I wouldn't say it is required.
'When provided once, only the 'ge' prefix is supported'

Reply:
if it's just a date then it defaults to eq and that is not supported. So no

Reply:
Looking at appointment: http://fhir.cerner.com/millennium/r4/scheduling/appointment/#parameters
```
* The `received` parameter may be provided:
** once with the `ge` prefix.
** twice with the lower value prefixed by `ge` and the higher value prefixed by `le`.
```
shoutout to Sarah

Do we want to do the same thing here?

Reply:
https://github.com/cerner/fhir.cerner.com/pull/607/commits/1287fa77649531cdaf02fcc169b1a67612536562#diff-cc6586eb639048bdea01208ce786612c64361622cd67bd53a5bf7253f27a840aL74